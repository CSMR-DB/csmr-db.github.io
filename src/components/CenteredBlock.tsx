import styled, { StyledComponent } from 'styled-components'
import { WidthProperty } from 'csstype'

interface ICenteredBlockProps {
  maxWidth?: WidthProperty<1>
}

export const CenteredBlock: StyledComponent<
  'div',
  any,
  ICenteredBlockProps,
  never
> = styled.div`
  max-width: ${({
    maxWidth = '48rem',
  }: ICenteredBlockProps): WidthProperty<1> => maxWidth};
  margin: 0 auto;
`
