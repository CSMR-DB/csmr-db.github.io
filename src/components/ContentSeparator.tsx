import styled, { StyledComponent } from 'styled-components'

export const ContentSeparator: StyledComponent<
  'div',
  any,
  {},
  never
> = styled.div`
  padding-bottom: 4rem;
`
