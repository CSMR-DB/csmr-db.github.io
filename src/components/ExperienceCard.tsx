import React from 'react'
import { BackgroundColorProperty } from 'csstype'
import { Card, CardHeaderIconWrapper, CardBody } from './Card'
import { Flex } from './Flex'
import { DynamicIcon } from './DynamicIcon'
import styled, { StyledComponent } from 'styled-components'
import { DateFormatted } from './DateFormatted'
import { theme } from '../data/theme'
import { Reveal, Tween } from 'react-gsap'

interface IExperienceCardProps {
  index: number
  name: string
  type: string
  icon: string
  dateStart: string
  dateEnd: string
  description: string
  backgroundColor: BackgroundColorProperty
}

const ExperienceCardHeader: StyledComponent<
  'header',
  any,
  {},
  never
> = styled.header`
  padding: 2em 2em 0em 2em;

  h2 {
    margin: 0;
  }
`

const ExperienceCardHeaderText: StyledComponent<
  'div',
  any,
  {},
  never
> = styled.div`
  padding: 0 2em;

  @media ${theme.breakpoints.max.smartphone} {
    padding: 0 0 0 1em;
  }
`

export const ExperienceCard: (props: IExperienceCardProps) => JSX.Element = ({
  // index,
  name,
  type,
  icon,
  dateStart,
  dateEnd,
  description,
  backgroundColor,
}: IExperienceCardProps): JSX.Element => (
  <Reveal>
    <Tween from={{ width: '8rem', height: '8rem' }} to={{ width: '100%', height: 'auto' }} duration={.75} ease={"power4"}>
      <Card>
        <ExperienceCardHeader>
          <Flex justifyContent="flex-start">
            <CardHeaderIconWrapper backgroundColor={backgroundColor}>
              <DynamicIcon path={icon} />
            </CardHeaderIconWrapper>
            <Tween from={{ display: 'none' }} delay={1}>
              <ExperienceCardHeaderText>
                <h1>
                  {name} ({type})
                </h1>
                <h2>
                  <DateFormatted dateString={dateStart} /> -{' '}
                  <DateFormatted dateString={dateEnd} />
                </h2>
              </ExperienceCardHeaderText>
            </Tween>
          </Flex>
        </ExperienceCardHeader>
        <CardBody>
          <Tween from={{ display: 'none' }} delay={1}>
            <main>
              <p>{description}</p>
            </main>
          </Tween>
        </CardBody>
      </Card>
    </Tween>
  </Reveal>
)
