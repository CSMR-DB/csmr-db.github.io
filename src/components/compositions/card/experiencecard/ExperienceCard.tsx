import React from 'react'
import { BackgroundColorProperty } from 'csstype'
import { Reveal } from 'react-gsap'

import { StaticDataManager } from '../../../../data/StaticDataManager'

import { Card } from '../Card'
import { Flex } from '../../../Flex'
import { DynamicIcon } from '../../DynamicIcon'
import { DateFormatted } from '../../../DateFormatted'
import { ExperienceCardHeader } from './ExperienceCardHeader'
import { ExperienceCardHeaderText } from './ExperienceCardHeaderText'
import { CardBody } from '../CardBody'
import { CardHeaderIconWrapper } from '../CardHeaderIconWrapper'

interface IExperienceCardProps {
  name: string
  body: string
  type: string
  icon: string
  dateStart: string
  dateEnd: string
  backgroundColor: BackgroundColorProperty
}

export function ExperienceCard({
  name,
  body,
  type,
  icon,
  dateStart,
  dateEnd,
  backgroundColor,
}: IExperienceCardProps): JSX.Element {
  return (
    <Reveal>
      <Card>
        <ExperienceCardHeader>
          <Flex $justifyContent="flex-start">
            <CardHeaderIconWrapper $backgroundColor={backgroundColor}>
              <DynamicIcon
                path={icon}
                dynamicIcons={StaticDataManager.dynamicIcons}
              />
            </CardHeaderIconWrapper>
            <ExperienceCardHeaderText>
              <h1>
                {name} ({type})
              </h1>
              <h2>
                <DateFormatted dateString={dateStart} /> -{' '}
                <DateFormatted dateString={dateEnd} />
              </h2>
            </ExperienceCardHeaderText>
          </Flex>
        </ExperienceCardHeader>
        <CardBody>
          <p>{body}</p>
        </CardBody>
      </Card>
    </Reveal>
  )
}
