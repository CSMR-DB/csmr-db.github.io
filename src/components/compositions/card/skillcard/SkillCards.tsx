import React from 'react'

import {
  ISkillsetAllMarkdownRemark,
  ISkillsetEdge,
} from '../../../../types/interfaces'
import { routeGenerator } from '../../../../utils/routeGenerator'

import { SkillCard } from './SkillCard'

export function SkillCards({ edges }: ISkillsetAllMarkdownRemark): JSX.Element {
  return (
    <>
      {edges.map(
        (
          { node: { frontmatter, fileAbsolutePath } }: ISkillsetEdge,
          i: number
        ): JSX.Element => (
          <SkillCard
            level={frontmatter.level || 0}
            title={frontmatter.title}
            description={frontmatter.excerpt}
            time={frontmatter.time}
            key={i}
            index={i}
            skillColor={frontmatter.skillColor || 'black'}
            fileAbsolutePath={routeGenerator(fileAbsolutePath)}
          />
        )
      )}
    </>
  )
}
