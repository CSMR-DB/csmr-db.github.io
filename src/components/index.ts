export { Absolute } from './Absolute'
export { ArticleBody } from './ArticleBody'
export { ArticleTags } from './ArticleTags'
export { Blend } from './Blend'
export {
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardHeaderFeaturedImage,
  CardHeaderIconWrapper,
  CardHeaderText,
} from './Card'
export { CardHeaderImageOrVideo } from './CardHeaderImageOrVideo'
export { CenteredBlock } from './CenteredBlock'
export { ClippedBtn } from './ClippedBtn'
export { ContentSeparator } from './ContentSeparator'
export { CTAButton } from './CTAButton'
export { DateFormatted } from './DateFormatted'
export { DesktopNav } from './DesktopNav'
export { DynamicIcon } from './DynamicIcon'
export { DynamicImage } from './DynamicImage'
export { ExperienceCard } from './ExperienceCard'
export { Excerpt } from './Excerpt'
export { Filter } from './Filter'
export { Fixed } from './Fixed'
export { Flex } from './Flex'
export { Footer } from './Footer'
export { Grid } from './Grid'
export { Header } from './Header'
export { Heart } from './Heart'
export { Image } from './Image'
export { ImageContainerCircle } from './ImageContainerCircle'
export { ImageDot } from './ImageDot'
export { Layout } from './Layout'
export { Logo } from './Logo'
export { MenuTile } from './MenuTile'
export { ProjectCard } from './ProjectCard'
export { ReadingTime } from './ReadingTime'
export { Relative } from './Relative'
export {
  ScrollIconWrapper,
  AnimatedScroller,
  ScrollText,
} from './ScrollIndicator'
export { SEO } from './SEO'
export { SkillCard } from './SkillCard'
export { SocialButton } from './SocialButton'
export { StyledIFrame } from './StyledIFrame'
export { StyledLink, StyledA } from './StyledLink'
export { WASDNav } from './WASDNav'
