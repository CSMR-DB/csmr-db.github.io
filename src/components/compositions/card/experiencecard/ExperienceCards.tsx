import React from 'react'

import {
  ExperienceMarkdownEdge,
  ExperienceAllMarkdownRemark,
} from '../../../../types/graphql.types'

import { ExperienceCard } from './ExperienceCard'

export function ExperienceCards({
  edges,
}: ExperienceAllMarkdownRemark): JSX.Element {
  return (
    <>
      {edges.map(
        (
          { node: { frontmatter } }: ExperienceMarkdownEdge,
          i: number
        ): JSX.Element => (
          <ExperienceCard
            name={frontmatter.title}
            type={frontmatter.type}
            description={frontmatter.excerpt}
            icon={`icons/${frontmatter.title.toLocaleLowerCase()}_logo.png`}
            backgroundColor={frontmatter.backgroundColor}
            dateStart={frontmatter.dateStart}
            dateEnd={frontmatter.dateEnd}
            key={i}
          />
        )
      )}
    </>
  )
}
