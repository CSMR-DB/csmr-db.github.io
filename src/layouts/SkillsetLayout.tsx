import React from 'react'

import {
  IExperienceAllMarkdownRemark,
  ISkillsetAllMarkdownRemark,
} from '../types/interfaces'

import { SEO } from '../components/compositions/SEO'
import { CenteredBlock } from '../components/CenteredBlock'
import { Grid } from '../components/Grid'
import { ContentSeparator } from '../components/ContentSeparator'
import { SkillCards } from '../components/compositions/card/skillcard/SkillCards'
import { ExperienceCards } from '../components/compositions/card/experiencecard/ExperienceCards'
import { ExperienceCard } from '../components/compositions/card/experiencecard/ExperienceCard'

export interface ISkillsetLayoutProps {
  skillsetData: ISkillsetAllMarkdownRemark
  experienceData: IExperienceAllMarkdownRemark
}

export function SkillsetLayout({
  skillsetData,
  experienceData,
}: ISkillsetLayoutProps): JSX.Element {
  return (
    <>
      <SEO
        title="Skillset"
        description="This is my trying to provide you with an indication about my capabilities & goals looking forward"
      />
      <CenteredBlock>
        <ContentSeparator>
          <h1>This is what I have worked with</h1>
          <p>
            A rough estimation of the total time I've spent on everything is
            included.
          </p>
        </ContentSeparator>
      </CenteredBlock>
      <ContentSeparator>
        <Grid columns="repeat(auto-fill, minmax(25rem, 1fr))" maxWidth="160rem">
          <SkillCards edges={skillsetData.edges} />
        </Grid>
      </ContentSeparator>
      <CenteredBlock>
        <ContentSeparator>
          <h1>Education</h1>
          <ExperienceCard
            name="Digital Media & Communication"
            type="HBO"
            icon="icons/hu_logo.jpg"
            dateStart="2015-09-05"
            dateEnd="2019-02-04"
            description="This is where my interest in programming started, as well as my first steps into graphic design. I learned about marketing and doing user research. Halfway through my final year I decided to drop out due to lack of motivation as I wanted to pursue learning more about programming."
            backgroundColor="#298CCD"
            index={0}
          />
        </ContentSeparator>
        <ContentSeparator>
          <h1>Professional experience</h1>
          <Grid>
            <ExperienceCards edges={experienceData.edges} />
          </Grid>
        </ContentSeparator>
      </CenteredBlock>
    </>
  )
}
