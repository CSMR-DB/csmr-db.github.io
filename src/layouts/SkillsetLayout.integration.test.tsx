import React, { Component } from 'react'
import { ThemeProvider } from 'styled-components'
import { create, ReactTestRenderer } from 'react-test-renderer'
import { mount, ReactWrapper } from 'enzyme'

import { StaticDataManager } from '../data/StaticDataManager'

import { SkillsetLayout } from './SkillsetLayout'
import { Layout } from '../components/Layout'
import { CenteredBlock } from '../components/CenteredBlock'
import { Grid } from '../components/Grid'
import { SkillCards } from '../components/compositions/card/skillcard/SkillCards'
import { ExperienceCard } from '../components/compositions/card/experiencecard/ExperienceCard'
import { ExperienceCards } from '../components/compositions/card/experiencecard/ExperienceCards'
import { ContentSeparator } from '../components/ContentSeparator'

describe('SkillsetLayout', (): void => {
  it('should match snapshot', (): void => {
    const comp: ReactTestRenderer = create(
      <ThemeProvider theme={StaticDataManager.theme}>
        <SkillsetLayout
          programmingData={StaticDataManager.skillsetData.allProgramming}
          graphicDesignData={StaticDataManager.skillsetData.allGraphicDesign}
          experienceData={StaticDataManager.experienceData.experienceData}
        />
      </ThemeProvider>
    )

    expect(comp).toMatchSnapshot()
  })

  it('should render internally', (): void => {
    const comp: ReactWrapper<any, Readonly<{}>, Component<{}, {}, any>> = mount(
      <ThemeProvider theme={StaticDataManager.theme}>
        <SkillsetLayout
          programmingData={StaticDataManager.skillsetData.allProgramming}
          graphicDesignData={StaticDataManager.skillsetData.allGraphicDesign}
          experienceData={StaticDataManager.experienceData.experienceData}
        />
      </ThemeProvider>
    )

    expect(comp.find(Layout).props().isLandingPage).not.toEqual(true)
    expect(comp.find(Layout).find(CenteredBlock).length).toEqual(3)
    expect(comp.find(Layout).find(Grid).length).toEqual(3)

    expect(
      comp.find(Layout).find(CenteredBlock).at(0).find('h1').text()
    ).toEqual('This is what I have developed with')
    expect(comp.find(Layout).find(Grid).at(0).props()).toMatchObject({
      $columns: 'repeat(auto-fill, minmax(25rem, 1fr))',
      $maxWidth: '160rem',
    })
    expect(comp.find(Layout).find(Grid).at(0).find(SkillCards).length).toEqual(
      1
    )
    expect(
      comp.find(Layout).find(Grid).at(0).find(SkillCards).props().edges
    ).toEqual(StaticDataManager.skillsetData.allProgramming.edges)

    expect(
      comp.find(Layout).find(CenteredBlock).at(1).find('h1').text()
    ).toEqual(expect.stringContaining('Graphic Design'))
    expect(comp.find(Layout).find(Grid).at(1).props()).toMatchObject({
      $columns: 'repeat(auto-fill, minmax(25rem, 1fr))',
      $maxWidth: '160rem',
    })
    expect(comp.find(Layout).find(Grid).at(1).find(SkillCards).length).toEqual(
      1
    )
    expect(
      comp.find(Layout).find(Grid).at(1).find(SkillCards).props().edges
    ).toEqual(StaticDataManager.skillsetData.allGraphicDesign.edges)

    expect(
      comp
        .find(Layout)
        .find(CenteredBlock)
        .at(2)
        .find(ContentSeparator)
        .at(0)
        .find('h1')
        .at(0)
        .text()
    ).toEqual('Education')
    expect(
      comp
        .find(Layout)
        .find(CenteredBlock)
        .at(2)
        .find(ContentSeparator)
        .at(1)
        .find('h1')
        .at(0)
        .text()
    ).toEqual('Professional experience')
    expect(comp.find(Layout).find(Grid).at(1).props()).toMatchObject({
      $columns: 'repeat(auto-fill, minmax(25rem, 1fr))',
      $maxWidth: '160rem',
    })

    expect(comp.find(Layout).find(ExperienceCard).first().props().name).toEqual(
      'Digital Media & Communication'
    )

    expect(comp.find(Layout).find(ExperienceCards).length).toEqual(1)
    expect(comp.find(Layout).find(ExperienceCards).props().edges).toEqual(
      StaticDataManager.experienceData.experienceData.edges
    )
  })
})
