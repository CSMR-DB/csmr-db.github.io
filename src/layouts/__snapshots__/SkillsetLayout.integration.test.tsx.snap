// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillsetLayout should match snapshot 1`] = `
.c1 {
  margin: 0 auto;
  padding: 1rem;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #f00;
}

.c1 a:hover {
  color: #d00;
}

.c0 {
  min-height: 100vh;
  background: #fff;
}

.c2 {
  margin: 0 auto;
  max-width: 48rem;
}

.c4 {
  display: grid;
  margin: 0 auto;
  height: auto;
  max-width: 160rem;
  gap: calc(1rem / 2);
  grid-template-columns: repeat(auto-fill,minmax(16rem,1fr));
  grid-template-rows: auto;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
}

.c22 {
  display: grid;
  margin: 0 auto;
  height: auto;
  max-width: auto;
  gap: calc(1rem / 2);
  grid-template-columns: repeat(auto-fill,minmax(16rem,1fr));
  grid-template-rows: auto;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
}

.c3 {
  padding-bottom: 4rem;
}

.c10 {
  fill: red;
  width: 1rem;
  height: 1rem;
}

.c14 {
  fill: blue;
  width: 1rem;
  height: 1rem;
}

.c16 {
  fill: green;
  width: 1rem;
  height: 1rem;
}

.c6 {
  -webkit-transition: all ease 0.5s;
  transition: all ease 0.5s;
  position: absolute;
  bottom: 1rem;
  right: 1rem;
  width: 100%;
  max-width: 8rem;
}

.c8 {
  padding: 2rem 2rem 0 2rem;
}

.c8 h1 {
  font-size: 1rem;
  font-weight: 100;
  text-align: right;
  -webkit-align-self: normal;
  -ms-flex-item-align: normal;
  align-self: normal;
}

.c8 h2 {
  font-size: 0.8rem;
  font-weight: 100;
}

.c13 {
  grid-area: 'footer';
  padding: 0 2rem 2rem;
}

.c7 {
  grid-area: 'header';
}

.c12 {
  grid-area: 'body';
  position: relative;
  padding: 2rem;
}

.c9 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  height: 100%;
  width: 100%;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: initial;
  -ms-flex-wrap: initial;
  flex-wrap: initial;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c19 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  height: 100%;
  width: 100%;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: initial;
  -ms-flex-wrap: initial;
  flex-wrap: initial;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c11 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  color: red !important;
}

.c11:hover {
  color: #d00;
}

.c11.active {
  color: #b00;
}

.c15 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  color: blue !important;
}

.c15:hover {
  color: #d00;
}

.c15.active {
  color: #b00;
}

.c17 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  color: green !important;
}

.c17:hover {
  color: #d00;
}

.c17.active {
  color: #b00;
}

.c5 {
  position: relative;
  height: 100%;
  border-radius: 0.25rem;
  display: grid;
  grid-template-areas: 'header' 'body' 'footer';
  grid-template-rows: auto 1fr auto;
  overflow: hidden;
  box-shadow: 0 0.075rem 0.075rem rgba(0,0,0,0.1), 0 0.125rem 0.125rem rgba(0,0,0,0.25);
  -webkit-transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  background: #fff;
}

.c5 header img {
  -webkit-transition: all 0.2s ease-in-out !important;
  transition: all 0.2s ease-in-out !important;
}

.c5 h1 {
  font-size: 1.1rem;
  margin: 0;
}

.c5 h2 {
  font-size: 1rem;
  font-style: italic;
  margin: 0;
  color: #aaa;
  margin-bottom: 1rem;
}

.c5 p {
  text-align: left;
}

.c18 {
  padding: 2rem 2rem 0rem 2rem;
}

.c18 h2 {
  margin: 0;
}

.c21 {
  padding: 0 0 0 1rem;
}

.c20 {
  border-radius: 999rem;
  overflow: hidden;
  width: 4rem;
  padding: 0.5rem;
  background: #298CCD;
}

.c23 {
  border-radius: 999rem;
  overflow: hidden;
  width: 4rem;
  padding: 0.5rem;
  background: blue;
}

.c24 {
  border-radius: 999rem;
  overflow: hidden;
  width: 4rem;
  padding: 0.5rem;
  background: red;
}

@media only screen and (min-width:600px) {
  .c1 {
    padding: 4rem 1rem;
  }
}

@media only screen and (min-width:600px) {
  .c4 {
    grid-template-columns: repeat(auto-fill,minmax(25rem,1fr));
    gap: 1rem;
  }
}

@media only screen and (min-width:600px) {
  .c22 {
    grid-template-columns: auto;
    gap: 1rem;
  }
}

@media only screen and (min-width:600px) {
  .c5 {
    text-align: justify;
  }
}

@media only screen and (min-width:1200px) {
  .c5:hover {
    box-shadow: 0 0.125rem 0.25rem rgba(0,0,0,0.25), 0 0.25rem 0.25rem rgba(0,0,0,0.2);
  }

  .c5:hover header img {
    -webkit-transform: scale(1.1);
    -ms-transform: scale(1.1);
    transform: scale(1.1);
  }
}

@media only screen and (min-width:600px) {
  .c21 {
    padding: 0 2rem;
  }
}

<div
  className="c0"
>
  <main
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <h1>
          This is what I have developed with
        </h1>
        <p>
          A rough estimation of the total time I've spent on everything is included.
        </p>
      </div>
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <article
          className="c5"
        >
          <div
            className="c6 backdrop"
          >
            
          </div>
          <header
            className="c7"
          >
            <div
              className="c8"
            >
              <div
                className="c9"
              >
                <svg
                  className="c10"
                  viewBox="0 0 315 342"
                >
                  <g
                    id="heart"
                  >
                    <path
                      d="M0 200 v-200 h200 a100,100 90 0,1 0,200 a100,100 90 0,1 -200,0 z"
                    />
                  </g>
                </svg>
                <h1>
                  <a
                    className="c11"
                    href="/skillset/react"
                  >
                    React
                  </a>
                </h1>
              </div>
            </div>
          </header>
          <main
            className="c12"
          >
            <p>
              I love JSX
            </p>
          </main>
          <footer
            className="c13"
          >
            <p>
              (time spent:
               
              <span>
                0
              </span>
               
              hours)
            </p>
          </footer>
        </article>
        <article
          className="c5"
        >
          <div
            className="c6 backdrop"
          >
            
          </div>
          <header
            className="c7"
          >
            <div
              className="c8"
            >
              <div
                className="c9"
              >
                <svg
                  className="c14"
                  viewBox="0 0 315 342"
                >
                  <g
                    id="heart"
                  >
                    <path
                      d="M0 200 v-200 h200 a100,100 90 0,1 0,200 a100,100 90 0,1 -200,0 z"
                    />
                  </g>
                </svg>
                <h1>
                  <a
                    className="c15"
                    href="/skillset/jest"
                  >
                    Jest
                  </a>
                </h1>
              </div>
            </div>
          </header>
          <main
            className="c12"
          >
            <p>
              It Jest needs to be done
            </p>
          </main>
          <footer
            className="c13"
          >
            <p>
              (time spent:
               
              <span>
                0
              </span>
               
              hours)
            </p>
          </footer>
        </article>
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <h1>
          I have also worked with Graphic Design
        </h1>
      </div>
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <article
          className="c5"
        >
          <div
            className="c6 backdrop"
          >
            
          </div>
          <header
            className="c7"
          >
            <div
              className="c8"
            >
              <div
                className="c9"
              >
                <svg
                  className="c16"
                  viewBox="0 0 315 342"
                >
                  <g
                    id="heart"
                  >
                    <path
                      d="M0 200 v-200 h200 a100,100 90 0,1 0,200 a100,100 90 0,1 -200,0 z"
                    />
                  </g>
                </svg>
                <h1>
                  <a
                    className="c17"
                    href="/skillset/affinity_designer"
                  >
                    Affinity Designer
                  </a>
                </h1>
              </div>
            </div>
          </header>
          <main
            className="c12"
          >
            <p>
              It has no proper geometry operations
            </p>
          </main>
          <footer
            className="c13"
          >
            <p>
              (time spent:
               
              <span>
                0
              </span>
               
              hours)
            </p>
          </footer>
        </article>
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <h1>
          Education
        </h1>
        <article
          className="c5"
        >
          <header
            className="c18"
          >
            <div
              className="c19"
            >
              <div
                className="c20"
              />
              <div
                className="c21"
              >
                <h1>
                  Digital Media & Communication
                   (
                  HBO
                  )
                </h1>
                <h2>
                  <span>
                    05 Sep 2015
                  </span>
                   -
                   
                  <span>
                    04 Feb 2019
                  </span>
                </h2>
              </div>
            </div>
          </header>
          <main
            className="c12"
          >
            <p>
              This is where my interest in programming started, as well as my first steps into graphic design. I learned about marketing and doing user research. Halfway through my final year I decided to drop out due to lack of motivation as I wanted to pursue learning more about programming.
            </p>
          </main>
        </article>
      </div>
      <div
        className="c3"
      >
        <h1>
          Professional experience
        </h1>
        <div
          className="c22"
        >
          <article
            className="c5"
          >
            <header
              className="c18"
            >
              <div
                className="c19"
              >
                <div
                  className="c23"
                />
                <div
                  className="c21"
                >
                  <h1>
                    React
                     (
                    Reactor
                    )
                  </h1>
                  <h2>
                    <span>
                      08 Jan 1994
                    </span>
                     -
                     
                    <span>
                      09 Jan 1994
                    </span>
                  </h2>
                </div>
              </div>
            </header>
            <main
              className="c12"
            >
              <p>
                JSX is just fantastic
              </p>
            </main>
          </article>
          <article
            className="c5"
          >
            <header
              className="c18"
            >
              <div
                className="c19"
              >
                <div
                  className="c24"
                />
                <div
                  className="c21"
                >
                  <h1>
                    Jest
                     (
                    Jester
                    )
                  </h1>
                  <h2>
                    <span>
                      08 Jan 1994
                    </span>
                     -
                     
                    <span>
                      09 Jan 1994
                    </span>
                  </h2>
                </div>
              </div>
            </header>
            <main
              className="c12"
            >
              <p>
                Jest a test
              </p>
            </main>
          </article>
        </div>
      </div>
    </div>
  </main>
</div>
`;
